server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -596,10 +601,22 @@ public static long parseToLong(
 
         @Override
         public Query distanceFeatureQuery(Object origin, String pivot, SearchExecutionContext context) {
+            failIfNotIndexedNorDocValuesFallback(context);
```

Comment:
were we silently not returning any result when the field was not indexed? Or failing later with some other exception?

Reply:
I think we were silently not returning any result...
We have this kind of bug in some other places unfortunately as well